File: CyberPunch GA/Population.py
import Individual

class Population:

    individuals = []

    def size():
        return len(individuals)

    def __init__(self, population_size, initialise):
        for i in range(population_size):
            individuals[i] = Individual()

        if initialise:
            for i in range(size()):
                newIndividual = new Individual()
                newIndividual.genIndividual()
                saveIndividual(i, newIndividual)

    def getIndividual(index):
        return individuals[index]

    def getFittest():
        fittest = individuals[0]

        for i in range(size()):
            if fittest.getFitness() <= getIndividual(i).getFitness():
                fittest = getIndividual(i)
        
        return fittest
    

    def save_individual(self, index, indiv):
        individuals[index] = indiv

File: CyberPunch GA/GeneticAlgorithm/GeneticAlgorithm/Population.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GeneticAlgorithm
{
    class Population
    {
        Individual[] individuals;

        /// <summary>
        /// Create a new population
        /// </summary>
        /// <param name="populationSize"></param>
        /// <param name="initialise"></param>
        public Population(int populationSize, Boolean initialise)
        {
            individuals = new Individual[populationSize];

            if (initialise)
            {
                for (int i = 0; i < Size(); i++)
                {
                    Individual newIndiv = new Individual();
                    newIndiv.GenIndividual();
                    SaveIndividual(i, newIndiv);
                }
            }
        }

        /// <summary>
        /// Return the individual at the given index
        /// </summary>
        /// <param name="index"></param>
        /// <returns></returns>
        public Individual GetIndividual(int index)
        {
            return individuals[index];
        }

        /// <summary>
        /// Gets the fittest gene in the population
        /// </summary>
        /// <returns></returns>
        public Individual GetFittest()
        {
            Individual fittest = individuals[0];

            for (int i = 0; i < Size(); i++)
            {
                if (fittest.GetFitness() <= GetIndividual(i).GetFitness())
                {
                    fittest = GetIndividual(i);
                }
            }
            return fittest;
        }

        /// <summary>
        /// 
        /// </summary>
        /// <returns></returns>
        public int Size()
        {
            return individuals.Length;
        }

        /// <summary>
        /// 
        /// </summary>
        /// <param name="index"></param>
        /// <param name="indiv"></param>
        public void SaveIndividual(int index, Individual indiv)
        {
            individuals[index] = indiv;
        }
    }
}
